packages/components/organisms/f-cookie-banner/src/components/CookieBanner.vue

```diff
@@ -0,0 +1,364 @@
+                    {{ $t('nonAcceptButtonText') }}
+                </button-component>
+            </div>
+        </div>
+        <legacy-banner
+            v-else
+            @hide-legacy-banner="hideBanner = true"
+        />
+    </div>
+</template>
+
+<script>
+import { globalisationServices } from '@justeat/f-services';
+import { VueGlobalisationMixin } from '@justeat/f-globalisation';
+
+import ButtonComponent from '@justeat/f-button';
+import '@justeat/f-button/dist/f-button.css';
+
+import LegacyBanner from './LegacyBanner.vue';
+
+import tenantConfigs from '../tenants';
+
+export default {
+    name: 'CookieBanner',
+
+    components: {
+        ButtonComponent,
+        LegacyBanner
+    },
+
+    mixins: [VueGlobalisationMixin],
+
+    props: {
+
+        locale: {
+            type: String,
+            default: 'en-GB'
+        },
+
+        isHidden: {
+            type: Boolean,
+            default: false
+        },
+
+        showLegacyBanner: {
+            type: Boolean,
+            default: false
+        },
+
+        cookieExpiry: {
+            type: Number,
+            default: 7776000
+        }
+    },
+
+    data () {
+        const locale = globalisationServices.getLocale(tenantConfigs, this.locale, this.$i18n);
+        const localeConfig = tenantConfigs[locale];
+        const theme = globalisationServices.getTheme(locale);
+
+        return {
+            config: { ...localeConfig },
+            tenantConfigs,
+            theme,
+            hideBanner: false,
+            legacyBanner: false
+        };
+    },
+
+    watch: {
+        isHidden (newVal) {
+            this.hideBanner = !!newVal;
+        },
+
+        showLegacyBanner (newVal) {
+            this.legacyBanner = !!newVal;
+        }
+    },
+
+    mounted () {
+        this.checkLegacyBannerFlag();
+        this.checkCookieBannerCookie();
+        this.focusOnTitle();
+    },
+
+    methods: {
+        /**
+         * Actions for "Accept all cookies" button
+         */
+        acceptActions () {
+            this.setCookieBannerCookie('full');
+            this.setLegacyCookieBannerCookie();
+            this.dataLayerPush('full');
+            this.hideBanner = true;
+        },
+        /**
+         * Actions for "Accept only required cookies" button
+         */
+        nonAcceptActions () {
+            this.setCookieBannerCookie('necessary');
+            this.setLegacyCookieBannerCookie();
+            this.dataLayerPush('necessary');
+            this.resendEvents();
+            this.removeUnnecessaryCookies();
+            this.hideBanner = true;
+        },
+        /**
+         * Set focus to the cookie consent banner title for accessibility
+         */
+        focusOnTitle () {
+            this.$refs.cookieBannerHeadingFocus.focus();
+        },
+        /**
+         * Check if the cookie banner has been shown to this user
+         */
+        checkLegacyBannerFlag () {
```

Comment:
We can probably rename this to something like `hasBannerBeenShownToUser` which would describe its intent better, I believe.

Reply:
This is checking the legacy banner flag in the JSON config, so the name seemed to make sense to me - happy to change it to whatever you want

Reply:
Just re-reading this, it is the comment that is wrong, and has been updated.